tools/src/bin/web-server.rs

```diff
@@ -94,20 +96,47 @@ fn handle(
     }
 
     let tree_name = &path[0];
-    let kind = &path[1];
+    let kind = path[1];
 
     println!("DBG {:?} {} {}", path, tree_name, kind);
 
-    match &kind[..] {
-        "rev" => {
+    match kind {
+        "rev" | "hgrev" => {
             if path.len() < 3 {
                 return not_found();
             }
 
-            let rev = &path[2];
+            let mut rev = path[2].to_string();
             let path = path.clone().split_off(3);
             let path = path.join("/");
 
+            if kind == "hgrev" {
+                let tree_config = cfg.trees.get(*tree_name).unwrap();
+                let git_path = match config::get_git_path(tree_config) {
+                    Ok(git_path) => git_path,
+                    Err(_) => return not_found(),
+                };
+                let output_result = Command::new("git")
+                    .arg("cinnabar")
+                    .arg("hg2git")
+                    .arg(&rev)
+                    .current_dir(&git_path)
+                    .output();
+                match output_result {
+                    Ok(output) if output.status.success() => {
+                        rev = git_ops::decode_bytes(output.stdout).trim().to_string();
+                    }
+                    Ok(output) => {
+                        println!("Cinnabar exit code: {:?}", output.status.code());
```

Comment:
`eprintln!` or something like that?

Reply:
Also may be worth to just format the whole `output` if possible, so that we have something more than the error code.